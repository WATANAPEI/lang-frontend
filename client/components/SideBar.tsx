import React from "react";
import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";
import Paper from "@material-ui/core/Paper";
import MenuList from "@material-ui/core/MenuList";
import MenuItem from "@material-ui/core/MenuItem";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    sideBar: {
      color: "red",
      margin: "0 5px",
      height: "100%"
    },
    li: {
      listStyle: "none"
    }
  })
);

interface Props {
  text: string;
}

function SideBar({ text }: Props) {
  const classes = useStyles();
  return (
    <Paper className={classes.sideBar}>
      <MenuList className={classes.li}>
        <MenuItem>{text}1</MenuItem>
        <MenuItem>{text}2</MenuItem>
      </MenuList>
    </Paper>
  );
}

export default SideBar;
