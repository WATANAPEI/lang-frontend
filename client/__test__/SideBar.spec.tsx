import React from "react";
import { createShallow, createMount } from "@material-ui/core/test-utils";
// @ts-ignore
import SideBar from "../components/SideBar.tsx";
import MenuList from "@material-ui/core/MenuList";
import MenuItem from "@material-ui/core/MenuItem";

describe("</SideBar />", () => {
  it("has a MenuList and MenuItems", () => {
    const testText = "sideBar text";
    const wrapper = createShallow()(<SideBar text={testText} />);
    expect(wrapper.find(MenuList)).toHaveLength(1);
    expect(wrapper.find(MenuItem)).toBeTruthy();
  });
  it("render received texts", () => {
    const testText = "sideBar text";
    const wrapper = createShallow()(<SideBar text={testText} />);
//    console.log(wrapper.text());
    expect(wrapper.text()).toEqual(
      expect.stringContaining(testText));
  });
});
