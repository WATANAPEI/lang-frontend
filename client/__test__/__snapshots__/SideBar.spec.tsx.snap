// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</SideBar /> has a MenuList and MenuItems 1`] = `
<WithStyles(ForwardRef(Paper))
  className="makeStyles-sideBar-1"
>
  <ForwardRef(MenuList)
    className="makeStyles-li-2"
  >
    <WithStyles(ForwardRef(MenuItem))>
      sideBar text
      1
    </WithStyles(ForwardRef(MenuItem))>
    <WithStyles(ForwardRef(MenuItem))>
      sideBar text
      2
    </WithStyles(ForwardRef(MenuItem))>
  </ForwardRef(MenuList)>
</WithStyles(ForwardRef(Paper))>
`;
